extensions/trace-propagators/src/test/java/io/opentelemetry/extension/trace/propagation/B3PropagatorTest.java

```diff
@@ -599,4 +600,62 @@ void extract_emptyCarrier() {
     assertThat(getSpanContext(b3Propagator.extract(Context.current(), emptyHeaders, getter)))
         .isEqualTo(SpanContext.getInvalid());
   }
+
+  @Test
+  void extract_DebugContext_SingleHeader() {
+    Map<String, String> carrier = new LinkedHashMap<>();
+    carrier.put(B3Propagator.COMBINED_HEADER, TRACE_ID + "-" + SPAN_ID + "-" + "d");
+
+    Context context = b3Propagator.extract(Context.current(), carrier, getter);
+    assertThat(getSpanContext(context))
+        .isEqualTo(
+            SpanContext.createFromRemoteParent(
+                TRACE_ID, SPAN_ID, SAMPLED_TRACE_OPTIONS, TRACE_STATE_DEFAULT));
+    assertThat(context.get(DEBUG_CONTEXT)).isEqualTo(Common.TRUE_INT);
+  }
+
+  @Test
+  void extract_DebugContext_MultipleHeaders() {
+    Map<String, String> carrier = new LinkedHashMap<>();
+    carrier.put(B3Propagator.TRACE_ID_HEADER, TRACE_ID);
+    carrier.put(B3Propagator.SPAN_ID_HEADER, SPAN_ID);
+    carrier.put(B3Propagator.DEBUG_HEADER, Common.TRUE_INT);
```

Comment:
I believe we're missing two test cases for when both sampled and debug are set. Especially sampled = false and debug = true is important.

Reply:
added